### ISSUE TITLE
Variants with a unit value are printed as the lowercased field name

### ISSUE THREAD

**Issue by nhweston** [inkytonik/cooma#46]
Example in the REPL:

```
cooma> < Foo = {} >
res0 : < Foo : Unit > = foo
```

I take it the intention is to print `true` and `false` for booleans.

For the reference backend, this seems to be caused by:

```
case VarR(FldR(c, `uniR`)) =>
    c.toLowerCase()
```

…in Interpreter.scala:288.

I can confirm this behaviour is also exhibited by the Truffle backend.

I noticed this while fixing tests after changing the return types of capability methods to variants. `Writer::write` now returns `< Left : String, Right : Unit >`. And here I noticed that programs returning `< Right : {} >` were having their results printed as `right`.

**Comment by inkytonik** [comment]
You've caught me :-)

Yes, it's to do with Booleans. To me it doesn't seem wrong to show unit values in this way, since they are essentially enumeration values and there is no other information to show.

We could hard-code it for Boolean, but that seems wrong since the prelude might not define Boolean.

We could use "True" and "Right" instead of "true" and "right" which would make the connection to the variant field name more apparent.

**Comment by nhweston** [comment]
> We could hard-code it for Boolean, but that seems wrong since the prelude might not define Boolean.

Yes, and also `true` and `false` are not reserved words. (And I don't think we want them to be.)

> We could use "True" and "Right" instead of "true" and "right" which would make the connection to the variant field name more apparent.

The problem is that `True`, `Right`, and any other name that could appear as a field name could also be a variable name. This could lead to confusion.

> Or we should just go back to showing the value "as it is", `<True = {}>` etc which is verbose but has the advantage of being precise.

I would prefer this to the two alternatives above. Perhaps a compromise is to omit the unit value and print it as `<True>`.

**Comment by inkytonik** [comment]
We'll go with the `<True>` form which should be supported as input as well.

**Comment by inkytonik** [comment]
We've decided to not make this change. Unit variant values should be printed in full, as in `<True = {}>` for consistency and avoiding special cases.

**Comment by inkytonik** [comment]
Closed by afee532e6ffe28f7526eb34bc7970c69a8bc4640
